botocore/utils.py

```diff
@@ -2938,8 +2938,8 @@ def _validate_allowed_url(self, full_url):
         is_whitelisted_host = self._check_if_whitelisted_host(parsed.hostname)
         if not is_whitelisted_host:
             raise ValueError(
-                "Unsupported host '%s'.  Can only "
-                "retrieve metadata from these hosts: %s"
+                "Unsupported host '%s'.  Can only retrieve metadata "
+                "from a loopback address or one of these hosts: %s"
                 % (parsed.hostname, ', '.join(self._ALLOWED_HOSTS))
```

Comment:
nit; when we update strings, we should try to move them towards either the `format()` or `f-string` style. There is however a single exception to this for log statements which are still optimized for this format.
```suggestion
                f"Unsupported host '{parsed.hostname}'.  Can only retrieve metadata from"
                f"a loopback address or one of these hosts: {', '.join(self._ALLOWED_HOSTS)}"
```